src/pip/_internal/index.py

```diff
@@ -252,6 +257,144 @@ def _get_html_page(link, session=None):
     return None
 
 
+class LinkCollector(object):
+
+    def __init__(
+        self,
+        session,       # type: PipSession
+        search_scope,  # type: SearchScope
+    ):
+        self.search_scope = search_scope
+        self.session = session
+
+    @property
+    def find_links(self):
+        # type: () -> List[str]
+        return self.search_scope.find_links
+
+    @staticmethod
+    def _sort_locations(locations, expand_dir=False):
```

Comment:
s/sort/categorize/ maybe?

Reply:
You read my mind. :) I changed this to `group_locations()` in the smaller setup PR. Does that name work? (I agree it's not sorting.)

Reply:
Yep yep!